[exhentai] use exhentai data in hitomi

chemPolonium opened the pull request:
Sorry for my poor python skill. Just to notify that metadata from exhentai is usable when the source is set to "hitomi". After this merge, metadata like "title_jpn" can be used when using "hitomi" source.

mikf commented:
You don't need to make any code changes for that, enabling `metadata` and `parent-metadata` for exhentai is enough, see https://github.com/mikf/gallery-dl/discussions/2066#discussioncomment-1838413

chemPolonium commented:
> You don't need to make any code changes for that, enabling `metadata` and `parent-metadata` for exhentai is enough, see [#2066 (reply in thread)](https://github.com/mikf/gallery-dl/discussions/2066#discussioncomment-1838413)

I tried this, but this only support e-hentai.org, when I use exhentai.org, it shows
`[exhentai][error] Unable to download data:  JSONDecodeError: [Errno Expecting value] : 0`

I've added metadata:true and parent-metadata:true in exhentai extractor setting, and metadata:false in hitomi extractor setting. Because it's exhentai.org, I also added username and password. When source set to gallery, metadata is right, metadata_from_page seems fine. But source set to hitomi, it uses metadata_from_api, which seems not work in exhentai.org

mikf commented:
Hmm, you are right, this doesn't work anymore for exhentai.org. Pretty sure it did when implementing and testing this a few months ago.

What does still work though, is making the same API call to e-hentai.org. Even for exh only galleries and even when not logged in. Add `"domain": "e-hentai.org"` to your exhentai settings and it will fetch metadata from e-hentai.

chemPolonium commented:
Yes, it works! Great thanks!
This is how I solved this (for who saw this later):
```
        "exhentai": {
            "domain": "e-hentai.org",
            "metadata": true,
            "parent-metadata": true,
            "source": "hitomi"
        },
        "hitomi": {
            "filename": "{num:>04}.{extension}",
            "metadata": false,
            "directory": [
                "{category}",
                "{title_jpn}"
            ]
        }
```
